qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -0,0 +1,195 @@
+from qiskit.result import Result
+from qiskit.transpiler import Target
+
+from qiskit_dynamics import Solver
+from qiskit_dynamics.pulse import InstructionToSignals
+
+from qiskit_experiments.exceptions import QiskitError
+from qiskit_experiments.test.utils import FakeJob
+
+
+class IQPulseBackend(BackendV2):
+    """Pulse Simulator abstract class"""
+
+    def __init__(self, static_hamiltonian, hamiltonian_operators, dt=0.1 * 1e-9, **kwargs):
+        """Hamiltonian and operators is the Qiskit Dynamics object"""
+        super().__init__(
+            None,
+            name="PulseBackendV2",
+            description="A PulseBackend simulator",
+            online_date=datetime.datetime.utcnow(),
+            backend_version="0.0.1",
+        )
+        self._defaults = PulseDefaults.from_dict({})
+        self.converter = None
+
+        self.static_hamiltonian = static_hamiltonian
+        self.hamiltonian_operators = hamiltonian_operators
+        self.solver = Solver(self.static_hamiltonian, self.hamiltonian_operators, **kwargs)
+        self._target = Target(dt=dt, granularity=16)
+        self.gound_state = np.zeros(self.solver.model.dim)
+        self.gound_state[0] = 1
+        self.y_0 = np.eye(self.solver.model.dim)
+
+    @property
+    def target(self):
+        return self._target
+
+    @property
+    def max_circuits(self):
+        return None
+
+    def defaults(self):
+        """return backend defaults"""
+        return self._defaults
+
+    @classmethod
+    def _default_options(cls):
+        return Options(shots=1024)
+
+    @staticmethod
+    def _get_info(inst):
+        p_dict = inst.operation
+        qubit = tuple(int(str(val)[-2]) for val in inst.qubits)
+        params = tuple(float(val) for val in p_dict.params)
+        return qubit, params, p_dict.name
+
+    @staticmethod
+    def _state_vector_to_result(
+        state: Union[Statevector, np.ndarray],
+        shots: int,
+        meas_return: MeasReturnType,
+        meas_level: MeasLevel,
+    ):
+        """Convert the state vector to IQ data or counts."""
+
+        # if meas_level == MeasLevel.CLASSIFIED:
+        # sample from the state vector. There might already be functions in Qiskit to do this.
+        #     counts = ...
+        # if meas_level == MeasLevel.KERNELED:
+        #     iq_data = ... create IQ data.
+        pass
+
+    def solve(self, schedule_blocks, qubits):
+        """Soleves a single schdule block and returns the unitary"""
+        if len(qubits) > 1:
+            QiskitError("TODO multi qubit gates")
+
+        signal = self.converter.get_signals(schedule_blocks)
+        time_f = schedule_blocks.duration * self.dt
+        result = self.solver.solve(
+            t_span=[0.0, time_f],
+            y0=self.y_0,
+            t_eval=[time_f],
+            signals=signal,
+            method="RK23",
+        ).y[0]
+
+        return result
+
+    # Parameters differ from overridden 'run' method
```

Comment:
I'm not able to close reviews as resolved, could you add me as an assigne to the PR?

Reply:
> Another thing that I forgot to comment. It would be good to have the "rz" gate in the basis gates. I.e. we need a unitary matrix for that as well. The single qubit backend should have the following as basis gates `["x", "sx", "rz"]`. `x` and `sx` should be implemented as simulated schedules and have a corresponding 3x3 unitary matrix while `rz(theta)` should just be implemented as `np.diag([1, np.exp(i*theta), np.exp(i*theta)])`. I'm not 100% sure what the phase of the second excited state should be for `rz`, this would be worth checking.

I remember $Z = [[1, 0, 0], [0, 0.5, 0], [0,0, -1]]$
So, 
$R_z(\theta) = exp(-\frac{i}{2} \theta Z) = [[e^{- 0.5 i \theta},  0, 0], [0, e^{- 0.25 i \theta}, 0], [0, 0, e^{0.5 i \theta}]]$
or
$R_z(\theta) = [[1,  0, 0], [0, e^{0.25 i \theta}, 0], [0, 0, e^{i \theta}]]$

@eggerdj right?